PollItem: Support checking associated socket

fwalch opened the pull request:
Adds is_socket and is_fd to PollItem to allow checking which socket a
PollItem has been created from.

Enables some use cases related to #68.

rotty commented on src/lib.rs:
```diff
@@ -1059,6 +1059,16 @@ impl<'a> PollItem<'a> {
+
+    /// Returns true if the polled socket is the given file descriptor.
+    pub fn is_fd(&self, fd: RawFd) -> bool {
+        self.fd == fd
```
This needs to check that `self.socket.is_null()`, or it may yield false positives.

fwalch commented:
Thanks for the comment, fixed the issue and added an assertion to the tests.

I'm not completely happy with the function names, as they are very similar to the revents checks (e.g. `is_readable`). I kept them because they were the shortest ones I could come up with. If you have any suggestions, I'd be happy to change them.

rotty commented:
Thanks for updating! I agree naming these `is_` is suboptimal, a it's commonly used for predicates, which take no arguments. What about `has_fd` and `has_socket`?

fwalch commented:
> What about `has_fd` and `has_socket`?

Renamed the functions.

rotty commented:
Thanks!
